### PR TITLE
FEATURE: S3 File Storage API Module

### PR DESCRIPTION
**Pull request by ivan-ivanovv**
- Created `upload-to-s3` API module to upload, download, list and delete files form and AWS S3 bucket
    - Supports large file uploads 
    - No local buffering -> directly streams the files to S3
- Cleaned up code base with unnecessary type annotations 
- Added unit tests for `response.ts` 
- Added test coverage configuration and run script 

Package dependencies: 
```
@aws-sdk/client-s3@3.676.0;  @aws-sdk/lib-storage@3.676.0; formidable@3.5.2
DEV: @smithy/util-stream@3.2.1; @types/formidable@3.4.5; aws-sdk-client-mock@4.1.0
```

### CODE REVIEW

**Comment by ivan-ivanovv** [comment]
@KDKHD Merging into development to be reviewed later

File: registry/modules/upload-to-s3/router/storeFileS3.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  .route("/files")
+  .get(protectedRoute, async (req, res, next) => {
+    const pageToken = req.query.pageToken as string;
+
+    await storeFileValidator()
+      .validateListFiles({ pageToken })
+      .then(storeFileController.getFileList)
+      .then((result) => res.json(response(result)))
+      .catch(next);
+  })
```
Comment (KDKHD) [inline registry/modules/upload-to-s3/router/storeFileS3.ts:37]: This exposes the S3 bucket to any authenticated user. Some additional security is missing here. 

Im not sure if we should create a router for the s3 module. Users need to carefully think about what parts of s3 bucket they want to expose.

File: registry/modules/upload-to-s3/utils/errors/storeFileS3.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      errorCode && "Error code: " + errorCode
+    }`,
+    400
+  );
+};
+
+export const s3UploadFailed = () => {
+  return new HttpError(
+    "S3FileUploadFailed",
+    `There was a problem when uploading your file to your S3 storage. Please check logs and try again`,
```
Comment (KDKHD) [inline registry/modules/upload-to-s3/utils/errors/storeFileS3.ts:16]: `Please check logs and try again`

The end customer will see this error. The end user will not be able to check logs.

File: registry/modules/upload-to-s3/utils/errors/storeFileS3.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    400
```
Comment (KDKHD) [inline registry/modules/upload-to-s3/utils/errors/storeFileS3.ts:17]: I think this should be a 5XX error.

File: registry/modules/upload-to-s3/utils/errors/storeFileS3.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { HttpError } from "@/modules/shared/utils/errors/HttpError.js";
+
+export const fileUploadFailed = (errorCode?: string) => {
+  return new HttpError(
+    "FileUploadFailed",
+    `There was an error uploading your file. ${
+      errorCode && "Error code: " + errorCode
+    }`,
+    400
```
Comment (KDKHD) [inline registry/modules/upload-to-s3/utils/errors/storeFileS3.ts:9]: Could be a 5XX error too.

File: registry/modules/upload-to-s3/utils/errors/storeFileS3.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    `There was a problem when uploading your file to your S3 storage. Please check logs and try again`,
+    400
+  );
+};
+
+export const errorDownloadingS3File = (fileName: string) => {
+  return new HttpError(
+    "ErrorDownloadingS3File",
+    `There was a problem when trying to download file ${fileName} from your S3 storage`,
+    404
```
Comment (KDKHD) [inline registry/modules/upload-to-s3/utils/errors/storeFileS3.ts:25]: 404 means Not Found. If the file does exist but downloading fails for an other reason, then I dont think 404 is a good status code.

**Pull request merged**